### ISSUE TITLE
Decayscore differs between UI and API

### ISSUE THREAD

**Issue by andurin** [MISP/MISP#5813]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu
| OS version (client)       | Ubuntu
| PHP version               | 7.0.33
| MISP version / git hash   | 2.4.124, 6bff2397406f5fee3d395e7fab98f14e4f3ab4fc
| Browser                   | Firefox

### Expected behavior

Decay Score query via API should be done in the same way and the same result as the UI.

### Actual behavior

Decay Score differs between both.

### Steps to reproduce the behavior

1. Enable a Decay model
2. Create an attribute with a datatype to match your model - for example "http://www.hazardous.com" as URL
2. Look at a Decay score via Web UI
3. Query the same with API

### Logs, screenshots, configuration dump, ...

Web UI

![2020-04-23 16-18-26](https://user-images.githubusercontent.com/541348/80187667-5280e880-8610-11ea-8a86-b70004cce1db.png)

API

```
curl \
 -d '{"returnFormat":"json","value":"http:\/\/www.hazardous.com","includeDecayScore":"true"}' \
 -H "Authorization: <<YOUR API KEY>>" \
 -H "Accept: application/json" \
 -H "Content-type: application/json" \
 -X POST https://misp/attributes/restSearch
```

Result (stripped):

```
        "value": "http://www.hazardous.com",
        "decay_score": [
          {
            "score": 99.942804841687,
            "decayed": false,
            "DecayingModel": {
              "id": "3",
              "name": "Phish"
            }
          }
        ],
```

**Comment by mokaddem** [comment]
I see three cases where you would end up in this situation:
1. You have multiple Attribute `http://www.hazardous.com` and are not viewing the result of the same Attribute
2. You have touched/modified/tagged the Attribute between the 2 score computations. What makes me thing of that is how high (99.94) the score is, indicating either that the last sighting/modification happened not so much ago or you have a very slow decaying model
3. We indeed have a bug in the system

Please could you check point 1 and 2 on your end while I'm looking at the 3rd one?
If it turns out that 1 and 2 are correct, would you mind sending me the decay model you are using as well as the target Attribute (sighting included)?
Thanks a lot!

**Comment by andurin** [comment]
1. It's my dev instance. Just did a attribute search in the UI - only one single attribute matching hazardous...
2. Sorry - also edit is also not the case. Just did a fresh "view". UI says 2.73, api says (5 seconds after ui view) 99.90095963625

The DecayModel I use would set the base-score to 25 because of a tag... so 99.xy is totally screwed.
You can redo it with the misp "default" phishing model for example. I just tried this to be independent from my possible modelling faults.

Using misp provided models:
UI
Phishing model                 9.85
API
Phishing model                 31.511815100744

**Comment by mokaddem** [comment]
Hey, thanks for the detailed feedback and analysis!
It was indeed a stupid bug.
(FYI, tags in restSearch where not passed to the computation function leading the `base_score` to fallback on the default value)
It's fixed here https://github.com/MISP/MISP/commit/eebde57ee10e5dad4cf6a7d7e25efd74a2612fe8

Also, I noted that you used the tag `retention:1d`. This tags is only supported by the decaying model if you use the `PolynomialExtended` formula, otherwise it's not considered.

Thanks again!

**Comment by andurin** [comment]
> It was indeed a stupid bug.

When did you see a clever bug the last time? :)

> (FYI, tags in restSearch where not passed to the computation function leading the `base_score` to fallback on the default value)
> It's fixed here [eebde57](https://github.com/MISP/MISP/commit/eebde57ee10e5dad4cf6a7d7e25efd74a2612fe8)

Fix acknowledged. Did a test and values are qual between Web and API now. Thanks for fixing it that fast.

> Also, I noted that you used the tag `retention:1d`. This tags is only supported by the decaying model if you use the `PolynomialExtended` formula, otherwise it's not considered.

I had to create a new Formula for my use case. It is a mix of Polynomial and PolynomialExtended to be able to have a default LifeTime for "untagged" attributes. A shorter (than the model) Lifetime for retention < Model Life Time and also the ability to extend the lifetime.
If one think it would be usefull for the rest of the world I would clean up the code and may PR it.

**Comment by mokaddem** [comment]
> When did you see a clever bug the last time? :)

Hehe

That's amazing! We also considered to extend the system to have dynamic lifetime (also based on tags and weight) but we wanted to have a feedback of the first version from the community.
If you took time to develop a new formula, I'm sure it's worth to have a look!
Feel free to open a PR! :)
